engine/server/sv_client.c

```diff
@@ -275,18 +275,8 @@ void SV_DirectConnect( netadr_t from )
 
 	BF_Init( &newcl->datagram, "Datagram", newcl->datagram_buf, sizeof( newcl->datagram_buf )); // datagram buf
 
-	// get the game a chance to reject this connection or modify the userinfo
-	if( !( SV_ClientConnect( ent, userinfo )))
-	{
-		if( *Info_ValueForKey( userinfo, "rejmsg" ))
-			Netchan_OutOfBandPrint( NS_SERVER, from, "%s\n%s\nConnection refused.\n", errorpacket, Info_ValueForKey( userinfo, "rejmsg" ));
-		else Netchan_OutOfBandPrint( NS_SERVER, from, "%s\nConnection refused.\n", errorpacket );
-
-		MsgDev( D_ERROR, "SV_DirectConnect: game rejected a connection.\n");
-		Netchan_OutOfBandPrint( NS_SERVER, from, "disconnect\n" );
-		SV_DropClient( newcl );
-		return;
-	}
+	// parse some info from the info strings (this can override cl_updaterate)
+	SV_UserinfoChanged( newcl, userinfo );
```

Comment:
Ensure that cl_updaterate will not be overwritten **after** SV_UserinfoChanged(as well, as other sv_client_t fields below).